client/rest/build.gradle

```diff
@@ -43,6 +43,21 @@ dependencies {
   testImplementation "org.elasticsearch:mocksocket:${versions.mocksocket}"
 }
 
+def generateVersionResource = tasks.register("generateVersionResource") {
```

Comment:
We could simplify all this by just putting a file under `src/main/resources` with contents like so:

```
elasticsearch-client=${elasticsearchVersion}
```

Then we can configure the `processResources` task to replace that value when copying the file to the resources output directory.

```
tasks.named("processResources").configure {
  expand elasticsearchVersion: versions.elasticsearch
}
```

Reply:
Excellent, TIL! Thanks!

I pushed 80c11f9 where I just `expand versions: versions` so that the placeholder in the properties file is the more familiar `${versions.elasticsearch}`. Can you please cross-check?

Reply:
That'll work.